meson.build

```diff
@@ -0,0 +1,81 @@
+project('argp-standalone', 'c')
+
+argp_source = files([
+	'argp-ba.c',
+	'argp-eexst.c',
+	'argp-fmtstream.c',
+	'argp-help.c',
+	'argp-parse.c',
+	'argp-pv.c',
+	'argp-pvh.c',
+	#'argp-test.c'
+])
+
+install_headers('argp.h')
+
+conf_data = configuration_data()
+
+cc = meson.get_compiler('c')
+
+conf_data.set10('HAVE_LIMITS_H', cc.check_header('limits.h'))
+conf_data.set10('HAVE_MALLOC_H', cc.check_header('malloc.h'))
+conf_data.set10('HAVE_UNISTD_H', cc.check_header('unistd.h'))
+conf_data.set10('HAVE_CONFIG_H', true)
+conf_data.set10('HAVE_LIBINTL_H', cc.check_header('libintl.h'))
+conf_data.set10('HAVE_ALLOCA_H', cc.check_header('alloca.h'))
+# conf_data.set10('_LIBC', true)
+
+conf_data.set10('HAVE_DECL_FLOCKFILE', cc.has_function('flockfile'))
+conf_data.set10('HAVE_DECL_FPUTS_UNLOCKED', cc.has_function('fputs_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_FPUTC_UNLOCKED', cc.has_function('fputc_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_FWRITE_UNLOCKED', cc.has_function('fwrite_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_PUTC_UNLOCKED', cc.has_function('putc_unlocked', prefix : '#include <stdio.h>'))
+
+conf_data.set10('HAVE_MEMPCPY', cc.has_function('mempcpy', prefix : '#include <string.h>'))
+conf_data.set10('HAVE_ASPRINTF', cc.has_function('asprintf', prefix: '#include <stdio.h>'))
+conf_data.set10('HAVE_STRCHRNUL', cc.has_function('strchrnul', prefix: '#include <string.h>'))
+conf_data.set10('HAVE_STRNDUP', cc.has_function('strndup', prefix: '#include <string.h>'))
+conf_data.set10('HAVE_STRCASECMP', cc.has_function('strcasecmp', prefix: '#include <string.h>'))
+
+conf_data.set10('HAVE_DECL_PROGRAM_INVOCATION_NAME',
+			    cc.has_function('program_invocation_name', prefix : '#include <errno.h>'))
+conf_data.set10('HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME',
+				cc.has_function('program_invocation_short_name', prefix : '#include <errno.h>'))
+
+deps = []
+if cc.check_header('libintl.h')
+	deps += dependency('intl')
+endif
```

Comment:
This probably doesn't work on musl. I think you should simply do `deps += dependency('intl', required: false)`, without even the conditional. What do you think?

Reply:
Why wouldn't it work? On musl it should resolve just like it does on glibc, via https://github.com/mesonbuild/meson/blob/a2934ca9d13ede4eb97b320bc768319ecad7b525/mesonbuild/dependencies/misc.py#L471-L476

Granted that checking for the header and then the dependency is essentially identical to your recommendation anyway. :p

Reply:
I see, intl is a built-in!

I was also thinking about the case where `dependency('intl')` makes it add some include path, meaning that `cc.check_header` fails but the header could still be found.